app/utils/logger.js

```diff
@@ -0,0 +1,58 @@
+// app/utils/logger.js
+
+const { createLogger, transports, format } = require('winston');
+const { combine, timestamp, printf, colorize, splat, simple } = format;
+const fs = require('fs');
+const path = require('path');
+require('winston-daily-rotate-file');
+
+// Ensure log directory exists
+const logDirectory = path.join(__dirname, '../../logs');
+fs.existsSync(logDirectory) || fs.mkdirSync(logDirectory);
+
+const logFormat = printf(({ level, message, timestamp, ...metadata }) => {
+    let msg = `${timestamp} [${level}]: ${message}`;
+    if (Object.keys(metadata).length) {
+        msg += ` ${JSON.stringify(metadata)}`;
+    }
+    return msg;
+});
+
+// Rotate file transport
+const rotateFileTransport = new transports.DailyRotateFile({
+    filename: path.join(logDirectory, 'app-%DATE%.log'),
```

Comment:
Please extract all the configurations into configuration files like .env.

In addition, 20m is quite a small size for logs, especially for production env.